packages/components/nodes/tools/Searxng/Searxng.ts

```diff
@@ -0,0 +1,119 @@
+                    'Comma separated list, specifies the active search categories. (see <a target="_blank" href="https://docs.searxng.org/user/configured_engines.html#configured-engines">Configured Engines</a>)',
+                optional: true,
+                additionalParams: true,
+                type: 'string'
+            },
+            {
+                label: 'Engines',
+                name: 'engines',
+                description:
+                    'Comma separated list, specifies the active search engines. (see <a target="_blank" href="https://docs.searxng.org/user/configured_engines.html#configured-engines">Configured Engines</a>)',
+                optional: true,
+                additionalParams: true,
+                type: 'string'
+            },
+            {
+                label: 'Language',
+                name: 'language',
+                description: 'Code of the language.',
+                optional: true,
+                additionalParams: true,
+                type: 'string'
+            },
+            {
+                label: 'Page No.',
+                name: 'pageno',
+                description: 'Search page number.',
+                optional: true,
+                additionalParams: true,
+                type: 'number'
+            },
+            {
+                label: 'Time Range',
+                name: 'time_range',
+                description:
+                    'Time range of search for engines which support it. See if an engine supports time range search in the preferences page of an instance.',
+                optional: true,
+                additionalParams: true,
+                type: 'string'
+            },
+            {
+                label: 'Safe Search',
+                name: 'safesearch',
+                description:
+                    'Filter search results of engines which support safe search. See if an engine supports safe search in the preferences page of an instance.',
+                optional: true,
+                additionalParams: true,
+                type: 'number'
+            }
+        ]
+        this.baseClasses = [this.type, ...getBaseClasses(SearxngSearch)]
+    }
+
+    async init(nodeData: INodeData, _: string): Promise<any> {
+        const apiBase = nodeData.inputs?.apiBase as string
+        const categories = nodeData.inputs?.categories as string
+        const engines = nodeData.inputs?.engines as string
+        const language = nodeData.inputs?.language as string
+        const pageno = nodeData.inputs?.pageno as number
+        const time_range = nodeData.inputs?.time_range as string
+        const safesearch = nodeData.inputs?.safesearch as 0 | 1 | 2 | undefined
+        const format = 'json' as 'json'
+
+        const params = {
+            format,
+            categories,
+            engines,
+            language,
+            pageno,
+            time_range,
+            safesearch
+        }
+
+        const headers = {}
```

Comment:
is this needed?

Reply:
Not yet but i wanted to keep it there to show the ability that this is there, if someone needs it and want to extend the node.